docs/Plugins_Guide.md

```diff
@@ -83,3 +74,80 @@ Ensure that the plugin is compiled on the same Ubuntu version (e.g., 2022.04) on
 Unity will fail load the plugin if the plugin was compiled against a newer version of `glibc` or `libstdc++` that is not present on the current system.
 
 Currently, the plugin is compatible with Ubuntu 22.04 or newer.
+
+## Development
+
+### Project Structure
+
+All plugin-related code can be found under the `plugins/` directory.
+Currently, there are two plugins:
+- `assignment/`: Implements different assignment algorithms using Google's OR-Tools library to assign interceptors to threats.
+- `protobuf/`: Implements loading Protobuf messages from Protobuf text files.
+
+Other useful code can be found in the other directories:
+- `base/`: Common base utilities.
+- `example/`: Contains an example plugin implementation.
+- `experimental/`: Experimental code, including toy examples to demonstrate various packages.
+
+### Status Codes
+
+Plugins should never throw exceptions as these cannot be caught by Unity and will cause the simulation to crash.
+Instead, plugins should always return a status code that Unity can check and handle appropriately.
+
+The complete list of status codes is defined in [`status.proto`](https://github.com/PisterLab/micromissiles-unity/blob/master/Assets/Proto/Plugin/status.proto) and is adapted from [Google's Abseil status codes](https://github.com/abseil/abseil-cpp/blob/master/absl/status/status.h).
+Defining a Protobuf enumeration for status codes allows both the C++ plugins and the Unity C# code to reference the same set of values.
```

Comment:
Would be nice to have a brief inline table of the most commonly used status codes (`OK, `INVALID_ARGUMENT`, `INTERNAL`) for quick/clear reference